drv/lpc55-spi-server/src/main.rs

```diff
@@ -117,172 +79,64 @@ fn main() -> ! {
-                    if !borrow_info.attributes.contains(LeaseAttributes::WRITE)
-                    {
-                        return Err(ResponseCode::BadArg);
-                    }
-
-                    datref.dat = Some(Transmit {
-                        task: caller,
-                        rpos: borrow_info.len,
-                        wpos: 0,
-                        len: borrow_info.len,
-                        op: Op::Read,
-                        rlease_num: 0,
-                        wlease_num: 0,
-                    });
-
-                    // Turning off receive without send is difficult (requires a
-                    // 16 bit write to a particular register) so just send some
-                    // bogus data for now
-                    datref.spi.drain();
-                    datref.spi.enable_tx();
-                    datref.spi.enable_rx();
-
-                    ringbuf_entry!(Trace::Start(Op::Read));
-                    Ok(())
-                }
-                Op::Exchange => {
-                    let ((), caller) =
-                        msg.fixed_with_leases(2).ok_or(ResponseCode::BadArg)?;
-
-                    if datref.dat.is_some() {
-                        return Err(ResponseCode::Busy);
-                    }
-
-                    let borrow_send = caller.borrow(0);
-                    let send_info =
-                        borrow_send.info().ok_or(ResponseCode::BadArg)?;
-                    if !send_info.attributes.contains(LeaseAttributes::READ) {
-                        return Err(ResponseCode::BadArg);
-                    }
-
-                    // Keep a minimum size to go with the FIFO settings
-                    if send_info.len < 8 {
-                        return Err(ResponseCode::BadArg);
-                    }
-
-                    let borrow_recv = caller.borrow(1);
-                    let recv_info =
-                        borrow_recv.info().ok_or(ResponseCode::BadArg)?;
-                    if !recv_info.attributes.contains(LeaseAttributes::WRITE) {
-                        return Err(ResponseCode::BadArg);
-                    }
-
-                    if recv_info.len != send_info.len {
-                        return Err(ResponseCode::BadArg);
-                    }
-
-                    datref.dat = Some(Transmit {
-                        task: caller,
-                        rpos: 0,
-                        rlease_num: 0,
-                        wpos: 0,
-                        wlease_num: 1,
-                        len: recv_info.len,
-                        op: Op::Exchange,
-                    });
-
-                    datref.spi.drain();
-                    datref.spi.enable_tx();
-                    datref.spi.enable_rx();
-
-                    ringbuf_entry!(Trace::Start(Op::Exchange));
-                    Ok(())
-                }
-            },
-        );
+        let _m = match sys_recv_closed(&mut [], 1, TaskId::KERNEL) {
```

Comment:
Since you're ignoring the values here you can likely simplify with 

```rust
if sys_recv_closed(&mut [], 1, TaskId::Kernel).is_err() {
    panic!()
}
```